Utilities/File.cpp

```diff
@@ -846,16 +846,24 @@ bool fs::truncate_file(const std::string& path, u64 length)
 #endif
 }
 
-bool fs::utime(const std::string& path, s64 atime, s64 mtime)
+bool fs::utime(const std::string& path, s64 atime, s64 mtime, std::optional<s64> ctime)
 {
 	if (auto device = get_virtual_device(path))
 	{
 		return device->utime(path, atime, mtime);
 	}
 
 #ifdef _WIN32
+	fs::stat_t stats;
+	if (!fs::stat(path, stats))
+	{
+		return false;
+	}
+
+	DWORD attr = stats.is_directory ? FILE_FLAG_BACKUP_SEMANTICS : FILE_ATTRIBUTE_NORMAL;
```

Comment:
Could be just FILE_FLAG_BACKUP_SEMANTICS | FILE_ATTRIBUTE_NORMAL without necessity to discriminate files and directories